documentation/guidelines.md

```diff
@@ -0,0 +1,303 @@
+- In cases where researchers truly disagree, they must each be able to express their viewpoint; this can include a personal "container" where the researcher can work out exactly how they would encode the data.
+
+
+## Agents
+
+For the sake of attribution, we distinguish between two types of agents: users and applications. Users represent real people, who can be cited as authors. Applications represent software that interacts with the graph.
+
+### Users
+
+In an application, it makes sense to let user nodes correspond one-to-one with users in an authentication system. In that case, the user nodes function as the "public" face of the user; this is the way their contributions are attributed.
+
+User nodes can have properties that describes the individual and their work. These should be defined by user themself.
+
+A user may have no informative properties at all. This allows users to make pseudo-nonymous contributions, where there is still a clear record of all contributions by that user.
+
+### Applications
+
+An application node represents a software application that makes contributions to the graph. These can be periodic updates or data migrations, which can be realised as annotations or direct changes.
+
+### Attribution
+
+Resources can be attributed to users or applications, but not both. However, it is not excluded that a resource may be attributed to multiple users (or multiple applications).
+
+## Collections
+
+Collections have two functions:
+
+- They present a collection of resources that a user has assembled
+- They are the context in which users make annotations
+
+In the most straightforward case, annotations are made on members of the collection. For example, a collection contains several books, and the owner of the collection has made annotations on those books.
+
+Note that even this case, context and membership are different things. Consider the following graph:
+
+```ttl
+_:GreatAuthors a edpopcol:Collection .
+_:Shakespeare rdfs:member _:GreatAuthors .
+
+_:a1  a edpopcol:Annotation ;
+    oa:hasTarget [ oa:hasSource _:Shakespeare ] .
+```
+
+While this may seem to imply that the annotation is made in the context of the `_:GreatAuthors` collection, it is possible that Shakespeare is a member of another collection, and the annotation was made in that context. Thus, we also need to specify:
+
+```ttl
+_:a1 as:context _:GreatAuthors .
+```
+
+It is not required that annotations target members of collections. It often makes sense to annotate resources that are indirectly related to members of the collection. In the example above, the user may want to annotate that Shakespeare's birthplace lies in England, by suggesting that `_:StratfordUponAvon _:country _:England` be added to the graph - but that triple is not directly related to the collection.
+
```

Comment:
Hm, true, you could separate them, but that could also introduce unnecessary complexity.

I assumed that our users would essentially be starting a project where they gather some records into a collection, and then make annotations on those records. However, the annotations could also be on related subjects (e.g. a collection of books, where you annotate the author of a book), which is why annotation targets are not formally linked to collection members.

They could also be decoupled; I can imagine uses for that, but I'm not sure what our users would want with it. It could also be confusing, since it requires users to have some sense of how they would use annotation context.

Reply:
Ah, I just got that wrong, I think.

I will change it so the context of annotations is a group of users, instead.

Reply:
Hm, thinking about this a bit more, I have the impression research groups are just an entirely different concept from what I was trying to express here.

By your description, I understand research groups mainly exist to restrict access. However, my intended use of `as:context` was more to describe something about where and why an annotation was made; from its documentation: "a means of grouping objects and activities that share a common originating context or purpose".

To me, access seems like another matter entirely. I did not think of the context as having any bearing on authorisation - rather, I saw it as a way to contextualise *public* annotations.

In any case, my impressions that annotations were relevant to collections was not correct. I've replaced this with a "project" class. Projects could be linked to user groups or even be one-to-one with them.

Perhaps this confusion happened because the graph currently has no expression of access at all. I assumed this would be handled by the SQL backend. But perhaps that would create cumbersome interactions, and groups and permissions should be included in the graph?